eldritch/stories.py

```diff
@@ -0,0 +1,402 @@
+    if isinstance(event, events.ForceTakeTrophy) and n_trophies >= 5:
+      return self.advance_story(owner, True)
+    if (
+        isinstance(event, events.GainOrLoss)
+        and (event.character == owner)
+        and (event.gains.get("dollars", 0) >= 5)
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class NunStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("Fear No Evil")
+
+  def get_interrupt(self, event, owner, state):
+    if isinstance(event, events.Curse) and event.character == owner:
+      return events.CancelEvent(event)
+    return super().get_interrupt(event, owner, state)
+
+  def get_usable_trigger(self, event, owner, state):
+    if isinstance(event, events.DiceRoll) and state.turn_phase == "upkeep" and not self.exhausted:
+      # Note: I see nothing that says it has to be one of her dice
+      choice = events.MultipleChoice(owner, "Choose a die to reroll", event.roll[:])
+      chosen = values.Calculation(choice, "choice_index")
+      return events.Sequence([
+          events.ExhaustAsset(owner, self),
+          choice,
+          events.RerollSpecificDice(event.character, event, chosen),
+      ], owner)
+    return None
+
+  def get_trigger(self, event, owner, state):
+    # Once per turn, not technically "exhaust to ..."
+    if (
+        self.exhausted
+        and isinstance(event, events.Mythos)
+        and event.is_done()
+    ):
+      return events.RefreshAsset(owner, self)
+    return None
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Bless(owner)
+
+
+class NunStoryFail(StoryResult):
+  def __init__(self):
+    super().__init__("I Shall Not Want")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.Sequence([
+        events.DiscardNamed(owner, "Curse"),
+        events.Bless(owner)
+    ], owner)
+
+
+class NunStory(Story):
+  def __init__(self):
+    super().__init__("He Is My Shepherd", "Fear No Evil", "I Shall Not Want")
+    self.max_tokens["clue"] = 2
+    self.tokens["clue"] = 0
+
+  def get_max_token_event(self, token_type, owner):
+    if token_type == "clue":
+      return self.advance_story(owner, True)
+    return events.Nothing()
+
+  def get_trigger(self, event, owner, state):
+    if isinstance(event, events.Bless):
+      return events.AddToken(self, "clue", owner)
+    if (
+        isinstance(event, events.BlessCurse)
+        and (event.character == owner)
+        and event.is_resolved()
+        and values.ItemNameCount(owner, "Curse").value(state) == 1
+    ):
+      return self.advance_story(owner, False)
+    return super().get_trigger(event, owner, state)
+
+
+class PhotographerStoryPass(StoryResult):
+  def __init__(self):
+    super().__init__("There's Your Proof")
+
+  def get_in_play_event(self, owner: characters.Character):
+    return events.DrawSpecific(owner, "specials", "Retainer")
```

Comment:
Have we set up retainers so that you can't get one if you already have one?

Reply:
I forget what they were at the time, but they are now "SelfDiscardingCard"s, which means that drawing one while you already have it should allow you to skip rolling on the turn after you get it.